astropy/modeling/tests/test_compound.py

```diff
@@ -181,6 +181,22 @@ def test_simple_two_model_class_compose_2d():
     r3 = R2(45, 45, 45, 45)
     assert_allclose(r3(0, 1), (0, -1), atol=1e-10)
 
+def test_n_submodels():
+    """
+    Test that CompoundModel.get_n_submodels() properly returns the number
+    of components.
+    """
+    g2 = Gaussian1D() + Gaussian1D()
```

Comment:
We should also test that non-compound model return 1, or is that non-applicable? @nden?